### PR TITLE
PYTHON-4672 Clarify Reauthentication and Speculative Authentication combination behavior

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/auth_oidc/test_auth_oidc.py [hunk 1]
```diff
@@ -1019,6 +1015,54 @@ def fetch(self, _):
...
+            client.test.test.insert_one({})
+
+        # Assert that the callback was called once.
+        self.assertEqual(self.request_called, 1)
+
+        # Assert there were `SaslStart` commands executed.
+        found = False
+        for event in listener.started_events:
+            if event.command_name.lower() != "saslstart":
+                found = True
```
Comment (Jibola) [inline test/auth_oidc/test_auth_oidc.py:1060]: should this be `event.command_name.lower() == "saslstart"`?

File: test/auth_oidc/test_auth_oidc.py [hunk 2]
```diff
@@ -1019,6 +1015,54 @@ def fetch(self, _):
...
+        assert found
```
Comment (Jibola) [inline test/auth_oidc/test_auth_oidc.py:1061]: NIT: (note I've still left it != for now)
```suggestion
        assert any([event.command_name.lower()  != "saslstart" for event in listener.started_events])
```

**Review by Jibola — CHANGES_REQUESTED** [review]

File: test/auth_oidc/test_auth_oidc.py (hunk 1 above)
Comment (blink1073) [reply test/auth_oidc/test_auth_oidc.py]: Yup, thanks!

**Review by Jibola — APPROVED** [review]
LGTM

**Pull request merged**